packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -11,6 +11,39 @@ import 'package:flutter/foundation.dart';
 import 'text_editing.dart';
 import 'text_input.dart';
 
+/// {@template flutter.services.textFormatter.maxLengthEnforcement}
+/// ### [MaxLengthEnforcement.enforced] versus
+/// [MaxLengthEnforcement.allowComposingTextToFinish]
+///
+/// Both [MaxLengthEnforcement.enforced]
+/// and [MaxLengthEnforcement.allowComposingTextToFinish] make sure the final length
+/// of the text does not exceed the max length specified.
+/// The difference is that [MaxLengthEnforcement.enforced] truncates the text
+/// whenever a new character is inserted into the text field,
+/// while [MaxLengthEnforcement.allowComposingTextToFinish] does not truncate when
+/// the user has unfinished composing text, allowing the "placeholder" text to
+/// temporarily exceed the max length limit, which provides a better
+/// user experience for entering ideographic characters (e.g., CJK characters)
+/// via composing on phonetic keyboards.
+///
+/// However, some input methods (Gboard for example) initiate text composition
+/// even for Latin characters. If the text field should only take
+/// Latin characters or numbers (e.g., verification code),
+/// [MaxLengthEnforcement.allowComposingTextToFinish] is generally a better fit.
+/// {@endtemplate}
+enum MaxLengthEnforcement {
+  /// Enforce max length on the editing value entirely, which is similar to
+  /// Gboard and iOS native IME.
```

Comment:
Maybe "Keep the length of the text input from exceeding the max length even when the text has unfinished composing region."